### PR TITLE
Added 150 * 4 dummy students for demo purposes

### PR DESCRIPTION
**Pull request by helloqx**
Added more students and modified test cases to be dynamic, instead of checking fixed list.

Dummy students are very basic:
>```
>matric number 'D'-X-YYYYYY-Z
>X: Year of student  
>YYYYYY: unique identifier  
>Z: Version identifier (B for this set)
>```

>Year `X` students will only take level `X` modules
>e.g. a year `1` student cannot take `CS2102` in current ay/sem, but may plan to take it in the next ay/sem

>Modules and Focus areas are **uniformly distributed**.

>Prereqs and Preclusions are not checked as it is assumed student plan info coming from frontend (NUS Oracle) will be checked on frontend side.

### CODE REVIEW

File: test/test_mod_taken_together_queries.py [hunk 1]
```diff
@@ -26,31 +26,37 @@ def test_query_module_taken_together_specified_mod(self):
             Tests querying the list of modules taken together
```
Comment (tgqiang) [inline test/test_mod_taken_together_queries.py]: Please remove the `__init__()` function; it is extremely redundant.

File: test/test_mod_taken_together_queries.py [hunk 2]
```diff
@@ -60,54 +66,70 @@ def test_query_module_taken_together_specified_mod_specified_aysem(self):
...
-                          'Discrete Structures', 2),
-                         ('CS2105', 'Introduction to Computer Networks', 'CS1010',
-                          'Programming Methodology', 2)]
+        required_list = [('AAA1111', 'Dummy 1', 'AAA1112', 'Dummy 2', 1)]
 
         assert_equal(len(list_of_mod_taken_together), len(required_list))
         assert_equal(sorted(list_of_mod_taken_together), sorted(required_list))
         assert_true(self.is_count_in_non_ascending_order(list_of_mod_taken_together))
 
         list_of_mod_taken_together = \
```
Comment (tgqiang) [inline test/test_mod_taken_together_queries.py]: This test case should be extracted out into another separate one. Not recommended to test different cases together in one test case.

File: test/test_modules_before_internship_backend.py [hunk 3]
```diff
@@ -17,26 +17,16 @@ class TestCode(object):
...
-
+        self.INTERN_MOD = 'CP3200'
+        self.INTERN_SEM = 'AY 17/18 Sem 1'
 
     def test_query_module_before_internship_empty(self):
         '''
             Tests querying the list of modules taken before internship,
             in AY/Sem which no one is doing internship in.
         '''
         list_of_modules_before_internship = model.get_mod_before_intern('AY 16/17 Sem 1')
```
Comment (tgqiang) [inline test/test_modules_before_internship_backend.py]: Can consider using your technique of adding a dummy entry, then testing if that entry should (or should not) be in a certain result set.

File: test/test_oversubscribed_modules_queries.py [hunk 4]
```diff
@@ -27,35 +27,20 @@ def test_query_oversubscribed_modules(self):
             Note to developers: Please check that you have repopulated
```
Comment (tgqiang) [inline test/test_oversubscribed_modules_queries.py]: Also, please remove the redundant `__init()__` function.

Comment (tgqiang) [inline test/test_oversubscribed_modules_queries.py]: Please revise the doc-string.

File: test/test_list_students_take_mod.py [hunk 5]
```diff
@@ -28,19 +28,22 @@ def test_list_students_take_module_one_focus(self):
...
-                         ['D5000001A', 2, 'Software Engineering', '-']
-                        ]
+        required_entry = ['dummyYr1A', 1, 'Database Systems', '-']
 
-        assert_equal(len(list_of_students_take_mod), len(required_list))
-        assert_equal(sorted(list_of_students_take_mod), sorted(required_list))
+        # Clean up database
+        model.delete_all_plans_of_student('dummyYr1A')
+        model.delete_student('dummyYr1A')
+        model.delete_student_focus_area('dummyYr1A')
```
Comment (nlzz22) [inline test/test_list_students_take_mod.py]: by right, you should delete_student_focus_area first before delete_student. Pls change this and add in a foreign key reference to the schema such that takesFocusArea's attribute nusnetId references foreign key from student's nusnetId

**Comment by helloqx** [comment]
@nlzz22 @tgqiang changed, please review again

**Comment by helloqx** [comment]
Made slight change to script so students have a small chance (<20%) of taking higher level modules, for better demo/illustration purposes:
image

**Comment by helloqx** [comment]
Made some changes to dummy students, now have small chance (<=20%) of taking a module one level higher for better illustration/demo purposes
image

**Pull request merged**